packages/contextual-help/src/ContextualHelp.ts

```diff
@@ -0,0 +1,205 @@
+    state,
+} from '@spectrum-web-components/base/src/decorators.js';
+import {
+    removeSlottableRequest,
+    SlottableRequestEvent,
+} from '@spectrum-web-components/overlay/src/slottable-request-event.js';
+import type { Placement } from '@spectrum-web-components/overlay/src/overlay-types.js';
+import {
+    IS_MOBILE,
+    MatchMediaController,
+} from '@spectrum-web-components/reactive-controllers/src/MatchMedia.js';
+
+import styles from './contextual-help.css.js';
+
+/**
+ * Spectrum Contextual help provides additional information about
+ * the state of either an adjacent component or an entire view.
+ * @element sp-contextual-help
+ *
+ * @slot Text content to display in the popover
+ * @slot link - link to additional informations
+ */
+export class ContextualHelp extends SpectrumElement {
+    protected isMobile = new MatchMediaController(this, IS_MOBILE);
+
+    public static override get styles(): CSSResultArray {
+        return [styles];
+    }
+
+    /**
+     * Optional title to be displayed inside the popover.
+     * @param {String} headline
+     */
+    @property()
+    public headline?: string;
+
+    /**
+     * Provides an accessible name for the action button trigger.
+     * @param {String} label
+     */
+    @property()
+    public label?: string;
+
+    /**
+     * The `variant` property applies specific styling on the action button trigger.
+     * @param {String} variant
+     */
+    @property()
+    public variant: 'info' | 'help' = 'info';
+
+    /**
+     * @type {"top" | "top-start" | "top-end" | "right" | "right-start" | "right-end" | "bottom" | "bottom-start" | "bottom-end" | "left" | "left-start" | "left-end"}
+     * @attr
+     */
+    @property({ reflect: true })
+    public placement?: Placement = 'bottom-start';
+
+    /**
+     * The `offset` property accepts either a single number, to
+     * define the offset of the Popover along the main axis from
+     * the action button, or 2-tuple, to define the offset along the
+     * main axis and the cross axis.
+     */
+    @property({ type: Number })
+    public offset: number | [number, number] = 0;
+
+    @state()
+    overlayOpen = false;
+
+    private get buttonAriaLabel(): string {
+        if (this.label) {
+            return this.label;
+        } else {
+            if (this.variant === 'help') {
+                return 'Help';
+            }
+            return 'Informations';
+        }
+    }
+
+    private renderPopover(event: SlottableRequestEvent): void {
+        event.stopPropagation();
+
+        if (event.data === removeSlottableRequest) {
+            render(undefined, event.target as HTMLElement);
+            return;
+        }
+
+        import('@spectrum-web-components/popover/sp-popover.js');
+
+        const template = html`
+            <sp-popover class="popover">
+                <section>
+                    ${this.headline &&
+                    html`
+                        <h2 class="heading">${this.headline}</h2>
```

Comment:
A user should have control over the heading level leveraged here. They should also have the ability to render HTML into this content.

Likely we want something like:
```
<div class="heading"> <!-- holds style delivery -->
  <slot name="heading"> <!-- normalize this name to that in other elements -->
    <h2>${this.heading}</h2> <!-- holds default headline level but reverts default heading styles -->
  </slot>
</div>
```
Or something.

Reply:
I actually did not use a slot _because_ of the fact that users have too much flexibility on the heading here and can actually use other things besides heading elements, such as `p` or `div`. I wanted to kind of enforce this one to be a heading. I can of course replace it with a slot.

Reply:
This is a really difficult balance to tread. We've been asked in the past by the accessibility team to ensure that patterns like this can accept multiple levels of heading to ensure the content is accessible, even if we haven't always gotten back to the related fixes 🙈. If there were other practical approaches to ensuring default delivery AND supporting varying levels of headlines, any new insight into the conversation that you might have would be greatly appreciated.

strictness vs support vs flexibility 😓

Reply:
I kind of like that as an option, though the work to do so is non-trivial. Might be worth creating an issue to come back to that at a later date?

Reply:
Sure, I'll create an issue for this. Do you see this being useful for other components as well, such as Dialogs?

Reply:
Rather than a dev mode warning I would like to get it surfaced up to the documentation site to have more visibility